lib/exp_operations.js

```diff
@@ -0,0 +1,103 @@
+// *****************************************************************************
+// Copyright 2013-2022 Aerospike, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// *****************************************************************************
+
+'use strict'
+
+/**
+ * @module aerospike/exp_operations
+ *
+ * @description This module provides functions to easily define operations to
+ * be performed on a record via the {@link Client#operate} command.
+ *
+ * @see {@link Client#operate}
+ *
+ * @example
+ *
+ * const Aerospike = require('aerospike')
+ * const op = Aerospike.operations
+ * const key = new Aerospike.Key('test', 'demo', 'mykey1')
+ *
+ * var ops = [
+ *   op.append('a', 'xyz'),
+ *   op.incr('b', 10),
+ *   op.read('b')
+ * ]
+ *
+ * Aerospike.connect((error, client) => {
+ *   if (error) throw error
+ *   client.put(key, { a: 'abc', b: 42 }, (error) => {
+ *     if (error) throw error
+ *     client.operate(key, ops, (error, record) => {
+ *       if (error) throw error
+ *       console.log(record) // => { b: 52 }
+ *       client.close()
+ *     })
+ *   })
+ * })
+ */
+
+const as = require('bindings')('aerospike.node')
+const ops = as.expOperations
+
+/**
+ * @class module:aerospike/expoperations~ExpOperation
+ *
+ * @classdesc Base class for all operations executed with the {@link
+ * Client#operate} command.
+ *
+ * Operations can be created using the methods in one of the following modules:
+ * * {@link module:aerospike/exp_operations} - General operations on all types.
+ * * {@link module:aerospike/operations} - General operations on all types.
+ * * {@link module:aerospike/lists} - Operations on CDT List values.
+ * * {@link module:aerospike/maps} - Operations on CDT Map values.
+ * * {@link module:aerospike/bitwise} - Operations on Bytes values.
+ */
+class ExpOperation {
+  /**
+   * @protected
+   */
+  constructor (op, bin, exp, flags, props) {
```

Comment:
`props` argument is never used and can be removed. Alternatively, you can skip the whole `ExpOperation` class and just use the base `Operation` class instead, and pass `exp` and `flags` as extra properties, i.e.

```javascript
exports.read = function (bin, exp, flags) {
   return new Operation(ops.READ, bin, { exp, flags })
}
```

Using a separate class to extend `Operation` really only makes sense if you need to add additional functionality like the `withContext` functions for the List and Map operations. That's not the case for the Expression operations.